includes/sanitizers/class-amp-gallery-block-sanitizer.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * Class AMP_Gallery_Block_Sanitizer.
+ *
+ * @package AMP
+ */
+
+/**
+ * Class AMP_Gallery_Block_Sanitizer
+ *
+ * Modifies gallery block to match the block's AMP-specific configuration.
+ */
+class AMP_Gallery_Block_Sanitizer extends AMP_Base_Sanitizer {
+
+	/**
+	 * Value used for width of amp-carousel.
+	 *
+	 * @since 1.0
+	 *
+	 * @const int
+	 */
+	const FALLBACK_WIDTH = 600;
+
+	/**
+	 * Value used for height of amp-carousel.
+	 *
+	 * @since 1.0
+	 *
+	 * @const int
+	 */
+	const FALLBACK_HEIGHT = 480;
+
+	/**
+	 * Tag.
+	 *
+	 * @var string Ul tag to identify wrapper around gallery block.
+	 * @since 1.0
+	 */
+	public static $tag = 'ul';
+
+	/**
+	 * Sanitize the gallery block contained by <ul> element where necessary.
+	 *
+	 * @since 0.2
+	 */
+	public function sanitize() {
+		$nodes     = $this->dom->getElementsByTagName( self::$tag );
+		$num_nodes = $nodes->length;
+		if ( 0 === $num_nodes ) {
+			return;
+		}
+
+		for ( $i = $num_nodes - 1; $i >= 0; $i-- ) {
+			$node = $nodes->item( $i );
+
+			// We're looking for <ul> elements that at least one child.
+			if ( 0 === count( $node->childNodes ) ) {
+				continue;
+			}
+
+			$attributes = AMP_DOM_Utils::get_node_attributes_as_assoc_array( $node );
+
+			// We are only looking for <ul> elements which have amp-carousel as class.
+			if ( ! isset( $attributes['class'] ) || false === strpos( $attributes['class'], 'amp-carousel' ) ) {
+				continue;
+			}
+
+			$images     = $node->getElementsByTagName( 'a' );
+			$num_images = $images->length;
+
+			if ( 0 === $num_images ) {
+				continue;
+			}
+
+			$attributes   = array(
+				'width'  => self::FALLBACK_WIDTH,
+				'height' => self::FALLBACK_HEIGHT,
```

Comment:
Not sure what could be a better alternative for setting width and height. Thoughts?

Reply:
Could the `AMP_Gallery_Block_Sanitizer` get added to the sanitizers list so that it is forced to be run after the `AMP_Img_Sanitizer`? Then you would be able to use the `width` and `height` of the images inside of the gallery to obtain the `width` and `height` for the overall gallery?

Reply:
Added `AMP_Gallery_Block_Sanitize::get_carousel_height()` which gets the heights of the images and sets the gallery height according to the smallest image. However, saw that basically all the images had the default height and width set (`400px`) within `AMP_Image_Sanitizer::determine_dimensions()`, so also added `AMP_Base_Sanitizer::set_attachment_width_and_height_by_src()` which gets the height and width from the image itself and is used in `AMP_Image_Sanitizer::determine_dimensions()` before the logic that adds the fallback width and height. Does this make sense, or does it interfere with the logic that sets the fallback width and height?

Reply:
Thinking that this might also change the looks for default gallery in an unexpected way.